src/test/java/com/example/postgresdemo/service/QuestionServiceTest.java

```diff
@@ -0,0 +1,161 @@
+package com.example.postgresdemo.service;
+
+import com.example.postgresdemo.exception.ResourceNotFoundException;
+import com.example.postgresdemo.model.Question;
+import com.example.postgresdemo.model.QuestionRequestDTO;
+import com.example.postgresdemo.model.QuestionResponseDTO;
+import com.example.postgresdemo.repository.QuestionRepository;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.NullSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.*;
+import org.springframework.data.domain.Page;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.any;
+
+@ExtendWith(MockitoExtension.class)
+class QuestionServiceTest {
+
+    @Mock
+    QuestionRepository questionRepository;
+
+    @InjectMocks
+    QuestionService questionService;
+
+    @Captor
+    ArgumentCaptor<Question> questionCaptor;
+
+    @Test
+    void testFindAll() {
+        Question question1 = new Question();
+        question1.setId(1L);
+        question1.setTitle("Title1");
+        question1.setDescription("Description1");
+        Question question2 = new Question();
+        question2.setId(2L);
+        question2.setTitle("Title2");
+        question2.setDescription("Description2");
+        Pageable pageable = PageRequest.of(0, 10);
+        Page<Question> page = new PageImpl<>(Arrays.asList(question1, question2));
+        Mockito.when(questionRepository.findAll(pageable)).thenReturn(page);
+
+        Page<QuestionResponseDTO> result = questionService.findAll(pageable);
+
+        Mockito.verify(questionRepository).findAll(pageable);
+        Assertions.assertEquals(2, result.getTotalElements());
+        Assertions.assertEquals(1L, result.getContent().get(0).getId());
+        Assertions.assertEquals("Title1\nDescription1", result.getContent().get(0).getBody());
+        Assertions.assertEquals(2L, result.getContent().get(1).getId());
+        Assertions.assertEquals("Title2\nDescription2", result.getContent().get(1).getBody());
+    }
+
+    @ParameterizedTest
+    @NullSource
+    @ValueSource(strings = {"", "Some Description"})
+    void testCreateWithDescriptionVariations(String description) {
```

Comment:
`@ValueSource` only allows passing single argument but you need to also pass an expected body to assert `result.getBody()` value. `@MethodSource` annotation better suits this purpose. Try using this helper method with combination MethodSource annotation:
```
    private static Stream<Arguments> provideDescriptions() {
        return Stream.of(
                Arguments.of(null, "Title\nnull"),
                Arguments.of("", "Title\n"),
                Arguments.of("Some description", "Title\nSome description")
        );
    }
```
Don't forget to add assertion for `result.getBody()`

Reply:
Fixed